tests/kani/FunctionAbstractions/mem_replace.rs

```diff
@@ -1,15 +1,36 @@
 // Copyright Kani Contributors
 // SPDX-License-Identifier: Apache-2.0 OR MIT
 
+//! Tests the `std::mem::swap` function using various function types.
+
 use std::mem;
 
+#[derive(PartialEq, Copy, Clone)]
+pub struct Pair {
+    value: u8,
+    key: u16,
+}
+
+unsafe impl kani::Invariant for Pair {
+    fn is_valid(&self) -> bool {
+        true
+    }
+}
+
+fn test<T: kani::Invariant + std::cmp::PartialEq + Clone>() {
+    let mut var1 = kani::any::<T>();
+    let var2 = kani::any::<T>();
+    let old_var1 = var1.clone();
+    assert_eq!(mem::replace(&mut var1, var2), old_var1);
+}
+
 #[kani::proof]
+#[kani::unwind(9)]
 fn main() {
-    let mut var1 = kani::any::<i32>();
-    let mut var2 = kani::any::<i32>();
-    let old_var1 = var1;
-    unsafe {
-        assert_eq!(mem::replace(&mut var1, var2), old_var1);
-    }
-    assert_eq!(var1, var2);
```

Comment:
Was this assert intentionally removed? Seems useful.

Reply:
Good catch. Readded